src/common/snippets/src/pass/split_dimension_m.cpp

```diff
@@ -0,0 +1,281 @@
+        } else if (const auto softmax_v8 = ov::as_type_ptr<ov::op::v8::Softmax>(op)) {
+            softmax_v8->set_axis(-1);
+        } else if (const auto softmax_v1 = ov::as_type_ptr<ov::op::v1::Softmax>(op)) {
+            softmax_v1->set_axis(softmax_v1->get_output_partial_shape(0).size()); // since new_shape.size() = old_shape.size() + 1
+        } else if (const auto broadcast = ov::as_type_ptr<ov::op::v1::Broadcast>(op)) {
+            // Broadcast is tokenized only between MatMuls -> Split M dimension
+            const auto shape_const = ov::as_type_ptr<ov::op::v0::Constant>(broadcast->input_value(1).get_node_shared_ptr());
+            OPENVINO_ASSERT(shape_const, "SplitDimensionM expects Broadcast with Constant output shape");
+            const auto new_shape = get_updated_shape(shape_const->cast_vector<size_t>(), broadcast->get_output_shape(0).size() - 2, true);
+            broadcast->set_argument(1, std::make_shared<ov::op::v0::Constant>(shape_const->get_element_type(), ov::Shape{new_shape.size()}, new_shape));
+        }
+    }
+
+    // Secondly, Update All M dimensions for remaining parameters
+    for (const auto& param : parameters) {
+        if (reshaped_params.count(param) == 0)
+            reshape_parameter(param, true);
+    }
+
+    // Update Transpose order on Result
+    for (const auto& res : results) {
+        const auto parent = res->get_input_node_shared_ptr(0);
+        if (ov::is_type<ov::op::v1::Transpose>(parent)) {
+            reshape_transpose(parent, false);
+        }
+    }
+
+    // Return the previous shape on outputs
+    for (size_t i = 0; i < subgraph->get_output_size(); ++i) {
+        const auto output_shape = subgraph->get_output_shape(i);
+        if (is_scalar(output_shape))
+            continue;
+
+        const auto& target_inputs = subgraph->get_output_target_inputs(i);
+        const auto shape_const = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{output_shape.size()}, output_shape);
+        const auto reshape = std::make_shared<ov::op::v1::Reshape>(subgraph->output(i), shape_const, false);
+        // Save output name
+        const auto original_output = body->get_results()[i]->get_input_node_shared_ptr(0);
+        const auto original_name = original_output->get_friendly_name();
+        reshape->set_friendly_name(original_name);
+        original_output->set_friendly_name(original_name + "_original");
+
+        for (const auto& input : target_inputs) {
+            input.replace_source_output(reshape);
+            // Result input tensor name was changed, the name has to be restored
+            if (ov::is_type<ov::op::v0::Result>(input.get_node())) {
+                input.get_tensor_ptr()->add_names(subgraph->output(i).get_tensor_ptr()->get_names());
+            }
+        }
+        subgraph->output(i).get_tensor_ptr()->set_names({});
+    }
+    subgraph->set_friendly_name(subgraph->get_friendly_name() + "_original");
```

Comment:
If reshape node is inserted after subgraph, it's better to set old subgraph's name to the reshape and update subgraph's name somehow to avoid names duplication. Names duplication might lead to model caching issues in some cases (when last node before result has the same name as other node)

In this case we append "_original" suffix - the same approach is used in LPT

Reply:
Ok, I see it now, thanks. Are these naming conventions summarized somewhere? If not, then maybe we should add a couple sentences in the Tokenizaton description.
My concern here is that node names are important (since the're exposed to exec graphs) and naming rules should be transparent for both customers and developers. Even for those who doesn't have extensive experience with LPT 😉

Reply:
@a-sidorova, do you think you could add snippets naming policy description?

Reply:
Discussed offline. Added snippets naming policy to tokenization pass.